### PR TITLE
Add oVirt platform

### PR DESCRIPTION
**Pull request by rgolangh**
This will add oVirt platform credentials support and is part of adding
oVirt platform to the installer as described in the enhancement document
openshift/enhancements#61

### CODE REVIEW

**Comment by dgoodwin** [comment]
@joelddiaz could you give this a review?

@rgolangh did you just install mockgen for this work or is it possible you have an older version laying around? Whichever of us is newer should be the one we stick with so if you did a fresh install for this Joel and I should update ours so we don't thrash on these changes.

File: pkg/assets/bindata.go [hunk 1]
```diff
@@ -231,37 +231,6 @@ func config_manager_metrics_service_yaml() ([]byte, error) {
...
-  annotations:
-    release.openshift.io/create-only: "true"
-data:
-  disabled: "true"
-`)
-
-func config_manager_operator_configmap_yaml() ([]byte, error) {
-	return _config_manager_operator_configmap_yaml, nil
-}
-
```
Comment (dgoodwin) [inline pkg/assets/bindata.go]: This looks suspicious... this is fairly new stuff being removed, wondering if there was a problem during devel on an older branch that leaked into your current PR?

**Comment by dgoodwin** [comment]
/hold

New feature missed 4.3, cant merge until 4.4.

File: pkg/assets/bindata.go (hunk 1 above)
Comment (rgolangh) [reply pkg/assets/bindata.go]: I cloned it today and I'm rebased on top of your merged commit 9b71397a.
What version of mockgen should I use?

Comment (joelddiaz) [reply pkg/assets/bindata.go]: it looks like on my machine i have golang-github-mock-1.2.0-3.fc31.x86_64 installed

Comment (rgolangh) [reply pkg/assets/bindata.go]: Thanks, that probably work, the PR is updated.

Do you want to document the mockgen version in the README?

File: pkg/ovirt/actuator.go [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+}
+
+func (a *OvirtActuator) syncCredentialSecret(cr *minterv1.CredentialsRequest, ovirtCreds *OvirtCreds, existingSecret *corev1.Secret, userPolicy string, logger log.FieldLogger) error {
+	sLog := logger.WithFields(log.Fields{
+		"targetSecret": fmt.Sprintf("%s/%s", cr.Spec.SecretRef.Namespace, cr.Spec.SecretRef.Name),
+		"cr":           fmt.Sprintf("%s/%s", cr.Namespace, cr.Name),
+	})
+
+	if existingSecret == nil || existingSecret.Name == "" {
+		if ovirtCreds == nil {
```
[the remark this answers is not in the document]
Comment (rgolangh) [reply pkg/ovirt/actuator.go]: I rather have that check instead of panic in the future for nil dereference. 
Or maybe I'll return a value and not a pointer.

File: pkg/ovirt/actuator.go [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+
+func (a *OvirtActuator) loadExistingSecret(cr *minterv1.CredentialsRequest) (*corev1.Secret, error) {
+	logger := a.getLogger(cr)
+
+	// Check if the credentials secret exists, if not we need to inform the syncer to generate a new one:
+	existingSecret := &corev1.Secret{}
+	err := a.Client.Get(context.TODO(), types.NamespacedName{Namespace: cr.Spec.SecretRef.Namespace, Name: cr.Spec.SecretRef.Name}, existingSecret)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			logger.Debug("secret does not exist")
```
[the remark this answers is not in the document]
Comment (rgolangh) [reply pkg/ovirt/actuator.go]: I want to create a new secret. I can use the err type NotFound for flow controll in the caller, right now I use the fact that the secret it empty. The handling is more generic like that I think.

File: pkg/ovirt/actuator.go [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+		return err
+	}
+
+	logger.Debugf("provisioning secret")
+	existingSecret, err := a.loadExistingSecret(cr)
+	if err != nil {
+		return err
+	}
+
+	err = a.syncCredentialSecret(cr, &ovirtCreds, existingSecret, "", logger)
```
[the remark this answers is not in the document]
Comment (rgolangh) [reply pkg/ovirt/actuator.go]: thanks for spotting, removed

File: pkg/ovirt/actuator.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (joelddiaz) [reply pkg/ovirt/actuator.go]: safe is better than sorry :) just wanted to make sure i wasn't missing some possibility that wasn't clear to me.

File: pkg/ovirt/actuator.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (joelddiaz) [reply pkg/ovirt/actuator.go]: help me understand the thinking when passing around a non-nil (but otherwise empty) secret.

i see in sync() we loadExistingSecret() and it can return a non-nil (but empty) secret if the secret isn't found.

then later in syncCredentialsSecret() it checks for both `existingSecret == nil || .Name == ""`. to me this could be simplified to just a check for `existingSecret == nil` if loadExistingSecret() returned a nil, nil. that way `existingSecret == nil` is the signal that there is no existing secret.

in summary, syncCredentialsSecret() doesn't use anything inside of existingSecret when it is empty, so why return the non-nil secret if syncCredentialsSecret() goes and creates a brand new secret object when there is no existing secret?

[the remark this answers is not in the document]
Comment (rgolangh) [reply pkg/ovirt/actuator.go]: Updating to a simpler version. PTAL

[the remark this answers is not in the document]
Comment (joelddiaz) [reply pkg/ovirt/actuator.go]: sgtm

File: pkg/assets/bindata.go (hunk 1 above)
Comment (joelddiaz) [reply pkg/assets/bindata.go]: i think this might be happening for a similar reason to why i had to make the second commit in PR #132 . once 132 merges, this PR should just need a simple rebase

**Comment by joelddiaz** [comment]
@rgolangh sorry for the slow reply (meetings all last week). can we also get some test cases written that mirror the kinds of tests we have for the other actuators / platforms?

File: pkg/ovirt/actuator_test.go [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+				Passord:  "",
+				CAFile:   "",
+			},
+			expectedToFail: true,
+		},
+	}
+
+	for _, v := range tests {
+		ovirtCreds, err := secretToCreds(&v.givenSecret)
+		if err != nil && !v.expectedToFail {
```
Comment (joelddiaz) [inline pkg/ovirt/actuator_test.go]: i think this can be simplified with
```
if v.expectedToFail {
 assert.Error(t, err, "expected failure")
} else {
  assert.NoError(t, err)
  assert.Equal(t, v.expectedCreds, ovirtCreds)
}
```

**Comment by rgolangh** [comment]
@joelddiaz should the do-not-merge/hold label still be set?
